django/contrib/admin/views/main.py

```diff
@@ -57,6 +67,9 @@ def __init__(self, request, model, list_display, list_display_links,
         self.sortable_by = sortable_by
 
         # Get search parameters from the query string.
+        change_list_form = ChangeListForm(request.GET)
+        if not change_list_form.is_valid():
+            messages.error(request, change_list_form.errors.as_text())
```

Comment:
The output here is suboptimal. e.g.`* p * Ensure this value is greater than or equal to 0`

We need to provide custom error messages that make sense when displayed as `messages`. We should maybe add the individual messages, in a loop, rather than the list.